tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -174,4 +178,307 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+
+		ginkgo.By("Verify PVC creation took longer than vpxd timeout")
+		gomega.Expect(elapsed > time.Second*time.Duration(vpxdTimeoutSecs)).To(
+			gomega.BeTrue(), "PVC creation was faster than vpxd timeout")
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		volumeID := pvs[0].Spec.CSI.VolumeHandle
+		storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyName)
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+		gomega.Expect(e2eVSphere.verifyDatastoreMatch(volumeID, sharedvmfsURL)).To(
+			gomega.BeTrue(), "volume %v was created on wrong ds", sharedvmfsURL)
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		}()
+		// TODO: Verify no orphan volumes are created
+	})
+
 })
+
+//fillVolumesInPods fills the volumes in pods upto given percentage
+func fillVolumeInPods(f *framework.Framework, pods []*v1.Pod) {
+	for _, pod := range pods {
+		size, err := getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		writeRandomDataOnPod(f, pod, size-66) // 66 is just from my observation and not technically derived
+	}
+}
+
+//writeRandomDataOnPod runs dd on the given pod and write count in Mib
+func writeRandomDataOnPod(f *framework.Framework, pod *v1.Pod, count int64) {
+	cmd := []string{"--namespace=" + pod.Namespace, "-c", pod.Spec.Containers[0].Name, "exec", pod.Name, "--",
+		"/bin/sh", "-c", "dd if=/dev/urandom of=/mnt/volume1/f1 bs=1M count=" + strconv.FormatInt(count, 10)}
+	_ = framework.RunKubectlOrDie(pod.Namespace, cmd...)
+}
+
+//setVpxdTaskTimeout sets vpxd task timeout to given number of seconds
+// Following cases will be handled here
+//  1. Timeout is not set, and we want to set it
+//  2. Timeout is set we want to clear it
+//  3. different timeout is set, we want to change it
+//  4. timeout is not set/set to a number, and that is what we want
+func setVpxdTaskTimeout(taskTimeout int) {
+	vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+	timeoutMatches := false
+	diffTimeoutExists := false
+
+	grepCmd := "grep '<timeout>' /etc/vmware-vpx/vpxd.cfg"
+	framework.Logf("Invoking command '%v' on vCenter host %v", grepCmd, vcAddress)
+	result, err := fssh.SSH(grepCmd, vcAddress, framework.TestContext.Provider)
+	if err != nil {
+		fssh.LogResult(result)
+		err = fmt.Errorf("couldn't execute command: %s on vCenter host %v: %v", grepCmd, vcAddress, err)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+
+	// we will replace the timeout value in the file if a different timeout was set previously
+	sshCmd := fmt.Sprintf(
```

Comment:
Where is this command used ?

Reply:
this will be used when timeout was not set before but we are setting it now